File: software/examples/rust/example_simple.rs
use std::{error::Error, io};

use tinkerforge::{ip_connection::IpConnection, ozone_bricklet::*};

const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your Ozone Bricklet.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let o = OzoneBricklet::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
                                          // Don't use device before ipcon is connected.

    // Get current ozone concentration.
    let ozone_concentration = o.get_ozone_concentration().recv()?;
    println!("Ozone Concentration: {} ppb", ozone_concentration);

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}

File: software/examples/go/example_simple.go
package main

import (
	"fmt"
	"github.com/tinkerforge/go-api-bindings/ipconnection"
	"github.com/tinkerforge/go-api-bindings/ozone_bricklet"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Ozone Bricklet.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	o, _ := ozone_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Get current ozone concentration.
	ozoneConcentration, _ := o.GetOzoneConcentration()
	fmt.Printf("Ozone Concentration:  ppb\n", ozoneConcentration)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()

}
